specification/profiling.md

```diff
@@ -24,8 +24,10 @@ data.
 
 ## Call Stack Sampling
 
-An instrumentation library that has profiling capabilities MUST be able to sample call stacks
-at a fixed interval. This default interval MUST default to 1 second.
+An instrumentation library that has profiling capabilities MUST be able to sample 
+call stacks at a fixed interval. This interval MUST be user-configurable at startup 
+via the `SPLUNK_PROFILER_CALL_STACK_PERIOD` environment variable. When not explicitly
```

Comment:
What unit is this value of the environment variable interpreted as? Seconds, milliseconds, microseconds ...

Should the unit be included in the variable name?

Reply:
In Java, the units are part of the value (`100ms` and `1s` are the same thing, the agent does duration parsing). It didn't feel like the right place to try and provide a full duration suffix spec tho, which is why I didn't specify. Do other agents also parse duration suffixes?

What do you think? Some options:
* Just change it to `SPLUNK_PROFILER_CALL_STACK_PERIOD_MS` and move on, skipping out on fancy (already built-in) duration parsing
* Specify that if units aren't provided it's assumed to be millis.

I'm thinking the latter.

Reply:
I like this convenience of java parsing, but probably we should follow the spec: https://github.com/open-telemetry/opentelemetry-specification/blob/2fb3b181ae2ee928039cda3baa8acc1eea701ce8/specification/sdk-environment-variables.md#duration

Unless we can guarantee that all our language can provide reach parsing.

Reply:
> * Specify that if units aren't provided it's assumed to be millis.

^^ this 👍 
This is how all duration config properties already work in Java.

Reply:
I think that defaulting to ms is in keeping with the spec that @iNikem linked to. To me, the fact that java provides supplemental helpful ways of parsing duration doesn't seem to be in conflict with the spec.

Reply:
It is not in conflict. See:
- https://github.com/open-telemetry/opentelemetry-specification/pull/2051#issuecomment-948957312
- https://github.com/open-telemetry/opentelemetry-specification/issues/1351

Reply:
How about renaming the env var `SPLUNK_PROFILER_CALL_STACK_INTERVAL`  so that it is similar to e.g. `OTEL_METRIC_EXPORT_INTERVAL`?

Reply:
> How about renaming the env var `SPLUNK_PROFILER_CALL_STACK_INTERVAL` so that it is similar to e.g. `OTEL_METRIC_EXPORT_INTERVAL`?

Well then that would break the original request to put the units in the env var name right?

Reply:
I do not think the unit should be in the name.   OTel recommends ms as the default unit and allows support for setting the unit explicitly

Reply:
Sounds good to me!